src/python/pants/pantsd/pants_daemon.py

```diff
@@ -225,6 +225,30 @@ def _setup_services(build_root, bootstrap_options, legacy_graph_scheduler, watch
         port_map=dict(pailgun=pailgun_service.pailgun_port),
       )
 
+  @staticmethod
+  def compute_invalidation_globs(bootstrap_options):
+    """
+    Combine --pythonpath and --pants_config_files(pants.ini) files that are in {buildroot} dir
+    with those invalidation_globs provided by users
+    :param bootstrap_options:
+    :return: A list of invalidation_globs
+    """
+    buildroot = get_buildroot()
+    invalidation_globs = []
+    globs = bootstrap_options.pythonpath + \
+      bootstrap_options.pants_config_files + \
+      bootstrap_options.pantsd_invalidation_globs
+
+    for glob in globs:
+      glob_relpath = os.path.relpath(glob, buildroot)
```

Comment:
I think that before attempting to relativize these, you should check `os.path.isabs`, and make the non-absolute paths absolute by `os.path.join`ing them to the buildroot.

After all that, you should be able to assume that they're all absolute and normalized, and _then_ you can safely check whether they relativize to the buildroot.

Reply:
^ this still needs addressing.

Reply:
Following method can handle/exclude both `../filename` and `/tmp/filename` (outside of buildroot)

```
    for glob in globs:
      glob_relpath = os.path.relpath(glob, buildroot)
      if glob_relpath and (not glob_relpath.startswith("../")):
        invalidation_globs.extend([glob_relpath, glob_relpath + '/**'])
      else:
        logging.getLogger(__name__).warning("Changes to {}, outside of the buildroot"
                                            ", will not be invalidated.".format(glob))
```

Reply:
Hm, yea... looks like it. That's a bit magical, but fine with me!